src/versions/2.0/json.ts

```diff
@@ -387,6 +387,47 @@ export async function validateJson(
   let warningCount = 0
   let errorCount = 0
 
+  const addErrorsToList = (validationErrors: ValidationError[]) => {
```

Comment:
Yeah, this one is kind of awkward because it has to modify the warning and error counts. I guess they could be passed in and then returned as an object, which would get assigned back to the local counts with a destructured assignment? That would let a function live in one place, probably a utils.ts like you mentioned. But that's probably better than having nearly the same local function in two places.